### PR TITLE
feat: implement taper challenge handling, and upate proofs

### PR DESCRIPTION
**Pull request by dignifiedquire**
Ref #38

### CODE REVIEW

**Comment by dignifiedquire** [comment]
@nicola @porcuquine need the default values for tapering, to finish this

File: src/bin/game.rs [hunk 1]
```diff
@@ -42,6 +42,18 @@ fn main() {
                 .default_value("10")
                 .takes_value(true),
         )
+        .arg(
+            Arg::with_name("taper")
+                .long("taper")
+                .help("fraction of challenges by which to taper at each layer")
+                .default_value("0.0"),
+        )
```
Comment (nicola) [inline src/bin/game.rs]: This should not be an option, tapered should be the default zigzag

File: src/bin/game.rs [hunk 2]
```diff
@@ -42,6 +42,18 @@ fn main() {
...
+        .arg(
+            Arg::with_name("taper-layers")
```
Comment (nicola) [inline src/bin/game.rs]: This should not be an option, tapered should be the default zigzag

File: src/models/leaderboard.rs [hunk 3]
```diff
@@ -112,8 +115,11 @@ impl Params {
                     challenge_count: val.challenge_count as i32,
                     vde: val.vde as i32,
                     degree: val.degree as i32,
-                    expansion_degree: val.expansion_degree.map(|v| v as i32),
-                    layers: val.layers.map(|v| v as i32),
+                    expansion_degree: val.zigzag.as_ref().map(|v| v.expansion_degree as i32),
+                    layers: val.zigzag.as_ref().map(|v| v.layers as i32),
+                    is_tapered: val.zigzag.as_ref().map(|v| v.is_tapered),
```
Comment (nicola) [inline src/models/leaderboard.rs]: we should never run a non-tapered zigzag

**Comment by porcuquine** [comment]
@dignifiedquire Default taper is 0.3.

**Comment by nicola** [comment]
see my comments ^

**Comment by nicola** [comment]
I will approve this only when this either becomes the only way to run zigzag or the default way (with updated readme)

**Comment by dignifiedquire** [comment]
@nicola @porcuquine updated to have fixed values for tapering and challenges, please recheck those values

**Comment by porcuquine** [comment]
@dignifiedquire I am sorry, it is supposed to be `1.0 / 3.0` not `0.3`. My mistake.

**Pull request merged**